lib/kubernetes-deploy/kubernetes_resource/stateful_set.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+require 'kubernetes-deploy/kubernetes_resource/pod_set_base'
+module KubernetesDeploy
+  class StatefulSet < PodSetBase
+    TIMEOUT = 5.minutes
+    ONDELETE = 'OnDelete'
+    attr_reader :pods
+
+    def sync
+      raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
+      @found = st.success?
+
+      if @found
+        stateful_data = JSON.parse(raw_json)
+        @status_data = stateful_data["status"]
+        @rollout_data = stateful_data["status"].slice("replicas")
+        @update_strategy = stateful_data["spec"]["updateStrategy"]["type"]
+        @status = @rollout_data.map { |state_replicas, num| "#{num} #{state_replicas.chop.pluralize(num)}" }.join(", ")
+        @pods = find_pods(stateful_data)
+      else # reset
+        @rollout_data = { "replicas" => 0 }
+        @status = nil
+        @pods = []
+      end
+    end
+
+    def deploy_succeeded?
+      if @update_strategy == ONDELETE
+        # Gem cannot monitor update since it doesn't occur until delete
+        unless @success_assumption_warning_shown
+          @logger.warn("WARNING: Your SS's updateStrategy is set to OnDelete, "\
+                       "which means updates will not be applied until its deleted. "\
+                       "If you are using k8s 1.7+, consider switching to RollingUpdates.")
+          @success_assumption_warning_shown = true
+        end
+        true
+      else
+        @status_data['current_generation'] == @status_data['observed_generation'] &&
+        @status_data['currentRevision'] == @status_data['updateRevision'] &&
+        @status_data['replicas'].to_i == @status_data['readyReplicas'].to_i
```

Comment:
According to the docs, `replicas` is the number created by the controller, which doesn't sound like the same thing as the number we ultimately want. I think we need to compare to `.spec.replicas` instead. (see my other comment though)